assets/src/design-system/components/mediaInput/index.js

```diff
@@ -0,0 +1,240 @@
+import PropTypes from 'prop-types';
+import { useState, forwardRef, useMemo } from 'react';
+import { __ } from '@web-stories-wp/i18n';
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import {
+  Button as DefaultButton,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+} from '../button';
+import { Landscape, Pencil } from '../../icons';
+import { Menu } from '../menu';
+import { Tooltip } from '../tooltip';
+import { MULTIPLE_VALUE } from '../../utils';
+import { MEDIA_VARIANTS } from './constants';
+
+const MediaRectangle = styled.section`
+  width: 64px;
+  height: 114px;
+  background-color: ${({ theme }) => theme.colors.bg.primary};
+  :focus {
+    outline: -webkit-focus-ring-color auto 1px;
+  }
+  border-radius: 4px;
+  position: relative;
+`;
+
+const MediaCircle = styled(MediaRectangle)`
+  border-radius: 100px;
+  height: 54px;
+  width: 54px;
+`;
+
+const ImageWrapper = styled.div`
+  border-radius: ${({ variant }) =>
+    variant === MEDIA_VARIANTS.CIRCLE ? 100 : 4}px;
+  overflow: hidden;
+  height: 100%;
+  display: flex;
+`;
+
+const DefaultImageWrapper = styled.div`
+  padding: 8px;
+`;
+
+const DefaultImage = styled(Landscape)`
+  width: 100%;
+  height: 100%;
+  display: block;
+  color: ${({ theme }) => theme.colors.standard.white};
+`;
+
+const Img = styled.img`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+`;
+
+const menuStyleOverride = css`
+  min-width: 100px;
+  margin-left: -50%;
+  li {
+    display: block;
+  }
+`;
+
+const Button = styled(DefaultButton)`
+  background-color: ${({ theme }) =>
+    theme.colors.interactiveBg.secondaryNormal};
+  position: absolute;
+  bottom: -8px;
+  right: -8px;
+  &:hover {
+    background-color: ${({ theme }) =>
+      theme.colors.interactiveBg.secondaryHover};
+  }
+`;
+
+const MediaOptions = {
+  [MEDIA_VARIANTS.RECTANGLE]: MediaRectangle,
+  [MEDIA_VARIANTS.CIRCLE]: MediaCircle,
+};
+
+const LoadingDots = styled.div`
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  width: 100%;
+  height: 100%;
+  pointer-events: none;
+  border-radius: 50%;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  display: flex;
+
+  &:after {
+    pointer-events: none;
+    color: ${({ theme }) => theme.colors.standard.white};
+    content: '.';
+    font-weight: bold;
+    animation: dots 1s steps(5, end) infinite;
+    margin-left: -12px;
+  }
+
+  @keyframes dots {
+    0%,
+    20% {
+      color: transparent;
+      text-shadow: 6px 0 0 transparent, 12px 0 0 transparent;
+    }
+    40% {
+      color: white;
+      text-shadow: 6px 0 0 transparent, 12px 0 0 transparent;
+    }
+    60% {
+      text-shadow: 6px 0 0
+          ${({ theme }) => theme.colors.standard.white},
+        12px 0 0 transparent;
+    }
+    80%,
+    100% {
+      text-shadow: 6px 0 0
+          ${({ theme }) => theme.colors.standard.white};
+        12px 0 0 ${({ theme }) => theme.colors.standard.white};
+    }
+  }
```

Comment:
I think we need to use `keyframes` [from S-C](https://styled-components.com/docs/api#keyframes) rather than this, as the above can lead to conflicts in naming.